semantics/src/main/kotlin/com/strumenta/kolasu/semantics/scoping/ScopeComputerCache.kt

```diff
@@ -0,0 +1,37 @@
+package com.strumenta.kolasu.semantics.scoping
+
+/**
+ * Abstract storage for scope computation results.
+ * @author Lorenzo Addazi <lorenzo.addazi@strumenta.com>
+ **/
+interface ScopeComputerCache {
```

Comment:
Same, what is the "scope" of the cache, how long does it live?

Reply:
I will improve documentation for all caches. The idea is that how long it lives really depends on the implementation and specific use case. The in-memory map implementation is volatile, but persistent caches could be implemented as well (for something like LSIF).

Reply:
The documentation of the _ScopeComputationCache_ is now:

```kotlin

/**
 * Abstract storage for storing scope computation results.
 *
 * Scope computation caches are meant to support the scope
 * computer not to repeat calculations. While a persistent
 * version of a scope computation cache can be implemented,
 * it is meant to be volatile by default.
 *
 * Persistent implementations should design dedicated
 * mechanisms for emptying, allocating and deallocating the cache.
 *
 **/

```